uberfire-extensions/uberfire-wires/uberfire-wires-core/uberfire-wires-core-grids/src/main/java/org/uberfire/ext/wires/core/grids/client/util/CellContextUtilities.java

```diff
@@ -113,6 +114,42 @@ public static GridBodyCellEditContext makeRenderContext(final GridWidget gridWid
                                            Optional.ofNullable(rp));
     }
 
+    public static void editSelectedCell(final GridWidget gridWidget) {
+        final GridData gridModel = gridWidget.getModel();
+
+        if (gridModel.getSelectedHeaderCells().size() > 0) {
```

Comment:
Hi @gitgabrio, thank you for feedback. Sorry I didn't respond sooner.

I am sorry, we can not forbid selection of multiple cells because of merged cells. They are rendered as one, but in fact, multiple cells are selected, see samples.

**Two selected header cells**
![screenshot from 2019-01-11 10-43-18](https://user-images.githubusercontent.com/8044780/51025971-bef7e580-158d-11e9-885e-c1d396ef0da2.png)

**Two selected cells**
![screenshot from 2019-01-11 10-44-53](https://user-images.githubusercontent.com/8044780/51026053-f8305580-158d-11e9-9e22-a68f8efa8175.png)


Similar holds true for `origin == null`. This is valid situation, as we register [KeyboardHandler](https://github.com/kiegroup/appformer/blob/master/uberfire-extensions/uberfire-wires/uberfire-wires-core/uberfire-wires-core-grids/src/main/java/org/uberfire/ext/wires/core/grids/client/widget/grid/impl/BaseGridWidgetKeyboardHandler.java) to the grid panel. It is valid that user (by mistake, by purpose) press key that represents some operation, but we won't execute that operation, because no cell is selected.

I agree with the size of the method, for better readability will split to private methods, commit is ongoing.

Reply:
@jomarko 
Good point, true.